### PR TITLE
scaffolding for pipeline react app

### PR DESCRIPTION
**Pull request by snopoke**
@bderenzi I threw this up to make it easier to start.

I decided to put it in a separate app because I think it might be more effort to migrate the old stuff then to start fresh, especially if we're migrating to langchain runnables. I'll try and get a POC of a runnable off the ground ASAP.

This now has the necessary components to start building:
* basic DB model
* save and load pipeline
* component list with drag and drop

Peek 2024-03-20 13-20

### CODE REVIEW

**Review by SmittieC — APPROVED** [review]
How cool is this! Looks like I need to clean out the cobwebs over react in my brain again

File: templates/pipelines/pipeline_builder.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+{% endblock %}
+{% block app %}
+  <div class="max-w-7xl mx-auto" id="pipelineBuilder">
+  </div>
+{% endblock %}
+{% block page_js %}
+  <script src="{% static 'js/pipeline-bundle.js' %}"></script>
+  <script type="module">
+    document.addEventListener('DOMContentLoaded', () => {
+      SiteJS.pipeline.renderPipeline("#pipelineBuilder", "{{ request.team.slug }}", 1);
```
[the remark this answers is not in the document]
Comment (snopoke) [reply templates/pipelines/pipeline_builder.html:16]: Yea, I haven't quite figured out the full flow and whether this screen should be the only way to edit a pipeline or if this is the 'builder' view and there is a plain form UI for the other bits (name, descriptions, default llm, etc). There aren't many other fields so maybe just doing it here is best.

**Comment by snopoke** [comment]
I'm going to merge this now since it's getting quite big. It is only accessible via the URL directly so shouldn't be an issue to deploy.

File: templates/pipelines/pipeline_builder.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (SmittieC) [reply templates/pipelines/pipeline_builder.html:16]: Cool cool. Just thinking out loud here. One could always have the builder view + form questions as a two-step process (build first then add description, or vice versa), or have a side panel with the form questions.

**Pull request merged**